liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -594,18 +594,38 @@ private void addArtifacts(org.eclipse.aether.graph.DependencyNode root, List<Fil
         List<Artifact> updatedArtifacts = new ArrayList<Artifact>();
         for (Dependency dep : dependencies) {
             boolean newDependency = true;
+
+            // match dependencies based on artifactId, groupId
             for (Dependency existingDep : existingDependencies) {
-                if (dep.getArtifactId().equals(existingDep.getArtifactId())) {
-                    newDependency = false;
-                    break;
+                if (dep.getArtifactId().equals(existingDep.getArtifactId())
+                        && dep.getGroupId().equals(existingDep.getGroupId())) {
+
+                    // check for matching version as long as version is not null in existing
+                    // dependency or new dependency
+                    if ((dep.getVersion() != null && dep.getVersion().equals(existingDep.getVersion()))
+                            || (dep.getVersion() == null)) {
```

Comment:
Should this also check if `existingDep.getVersion() == null` ?

Reply:
If `existingDep.getVersion() == null` `dep.getVersion().equals(existingDep.getVersion())` will return false.

Reply:
This could happen though: `dep.getVersion() == null && existingDep.getVersion() == null`.  Regardless, see my other comment about using an existing method to do the comparison instead.